synthesizer/src/store/helpers/memory/internal/mod.rs

```diff
@@ -147,6 +150,12 @@ impl<
     fn atomic_checkpoint(&self) {
         // Push the current length of the atomic batch to the checkpoint stack.
         self.checkpoint.lock().push_back(self.atomic_batch.lock().len());
+
+        // Increment the checkpoint depth, if it is set.
+        let mut checkpoint_depth = self.checkpoint_depth.lock();
+        if let Some(depth) = *checkpoint_depth {
+            *checkpoint_depth = Some(depth.saturating_add(1));
```

Comment:
what's the relationship between the length of `checkpoint` and `checkpoint_depth`? is it not the same?